### PR TITLE
the config files to setup pyenv locally and globally were added

### PR DESCRIPTION
**Pull request by UjjwalBudha**
I have uploaded the code for setting up pyenv and installing pyenv locally and globally using Ansible as well as i have added terraform config files for creating a ec2 instance.

### CODE REVIEW

File: ansible/roles/pyenv/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
+---
+- name: Install pyenv dependencies
```
Comment (milandangol) [inline ansible/roles/pyenv/tasks/main.yml:2]: Lets use variable and for loop concept for this

File: ansible/roles/pyenv/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    state: present
+    update_cache: true
+
+- name: Download pyenv script
+  ansible.builtin.get_url:
+    dest: /tmp/pyenv.sh
+    url: https://pyenv.run
+    mode: "0555"
+
+- name: Execute pyenv install script
```
Comment (milandangol) [inline ansible/roles/pyenv/tasks/main.yml:31]: lets not use shell  module for this. Please use script module

File: ansible/roles/pyenv/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  shell:
+    cmd: /tmp/pyenv.sh
+    creates: "{{ home_dir }}/.pyenv"
+
+# - name:
```
Comment (milandangol) [inline ansible/roles/pyenv/tasks/main.yml:36]: if not needed lets delet it

File: ansible/roles/pyenv/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    path: /root/test_pyenv
+    state: directory
+
+- name: Check if Python 3.7.7 is already installed
+  shell: "{{ pyenv_dir }} versions | grep -q '3.7.7'"
+  register: python_3_7_7_installed
+  changed_when: false
+  ignore_errors: true  
+
+- name: Check if Python 3.10.4 is already installed
```
Comment (milandangol) [inline ansible/roles/pyenv/tasks/main.yml:56]: lets not hardcode versions like this use variable

**Review by milandangol — CHANGES_REQUESTED** [review]

File: ansible/roles/pyenv/tasks/main.yml (hunk 3 above)
Comment (UjjwalBudha) [reply ansible/roles/pyenv/tasks/main.yml]: deleted the commented block of the code

File: ansible/roles/pyenv/tasks/main.yml (hunk 2 above)
Comment (UjjwalBudha) [reply ansible/roles/pyenv/tasks/main.yml:14]: included script module to execute pyenv script

File: ansible/roles/pyenv/tasks/main.yml (hunk 4 above)
Comment (UjjwalBudha) [reply ansible/roles/pyenv/tasks/main.yml:30]: created separate variables for the different versions

File: ansible/roles/pyenv/tasks/main.yml (hunk 1 above)
Comment (UjjwalBudha) [reply ansible/roles/pyenv/tasks/main.yml]: used variables to install all required dependencies

**Review by milandangol — APPROVED** [review]

**Pull request merged**